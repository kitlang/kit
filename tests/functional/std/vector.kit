function main() {
    var v: Vector[Int] = Vector.new(4);
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    v.push(1);
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);
    printf(if v.pop().isSome() then "some\n" else "none\n");
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);
    printf(if v.pop().isSome() then "some\n" else "none\n");
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    v.push(11);
    v.push(22);
    v.push(33);
    v.push(44);
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    for i in v {
        printf("iterate: %i\n", i);
    }

    v.push(55);
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    v.push(66);
    v.push(77);
    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    for i in v {
        printf("iterate: %i\n", i);
    }

    v.pop();
    v.pop();
    v.pop();

    printf("length = %zu\n", v.length);
    printf("capacity = %zu\n", v.data.length);

    for i in v {
        printf("iterate: %i\n", i);
    }

    printf("length = %zu\n", v.length);
    v.clear();
    printf("length after clear = %zu\n", v.length);

    v.free();

    var v2: Vector[Int] = Vector.new();
}
