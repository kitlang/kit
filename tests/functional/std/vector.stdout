length = 0
capacity = 4
length = 1
capacity = 4
some
length = 0
capacity = 4
none
length = 0
capacity = 4
length = 4
capacity = 4
iterate: 11
iterate: 22
iterate: 33
iterate: 44
length = 5
capacity = 6
length = 7
capacity = 9
iterate: 11
iterate: 22
iterate: 33
iterate: 44
iterate: 55
iterate: 66
iterate: 77
length = 4
capacity = 9
iterate: 11
iterate: 22
iterate: 33
iterate: 44
length = 4
length after clear = 0
